pkg/controllers/rollout/controller.go

```diff
@@ -329,6 +352,11 @@ func pickBindingsToRoll(allBindings []*fleetv1beta1.ClusterResourceBinding, late
 	klog.V(2).InfoS("Calculated the max number of bindings to remove", "clusterResourcePlacement", klog.KObj(crp),
 		"maxUnavailableNumber", maxUnavailableNumber, "minAvailableNumber", minAvailableNumber,
 		"lowerBoundAvailableBindings", lowerBoundAvailableNumber, "maxNumberOfBindingsToRemove", maxNumberToRemove)
+
+	// we can still update the bindings that are failed to apply already regardless of the maxNumberToRemove
+	for i := 0; i < maxUnavailableNumber && i < len(applyFailedUpdateCandidates); i++ {
```

Comment:
Hi Ryan! Apologies if I misunderstand anything, but would this still get stuck if, say, all bindings are bound but failed to get applied (-> maxNumberToRemove <= 0) + maxUnavailable set to 0 with a non-zero maxSurge?